### DISCUSSION TITLE
blog/posts/2022/10/28/linux-nanopi-r4s/

### DISCUSSION THREAD

**Discussion by giscus** [kohlschuetter/blog#1] [Giscus]
# blog/posts/2022/10/28/linux-nanopi-r4s/

This little ARM64 single-board computer dragged me into the world of device trees, voltage regulators, and one of the worst Heisenbugs I encountered so far.

https://kohlschuetter.github.io/blog/posts/2022/10/28/linux-nanopi-r4s/

**Comment by rbanffy** [comment]
I've been thinking of getting a couple of these gizmos for a small cluster project and it's good to know 6.1 will have upstream support thanks to you.

**Comment by obsidience** [comment]
I'm sure it will help.  Thank you for your service!

**Comment by midzelis** [comment]
Great write up! I've used a bunch of these little boxes. I have a NanoPi R1 and also a NanoPi NEO3. I found out about the https://www.armbian.com/ project, and they also do a lot to maintain a relatively recent Debian distro for these SBCs - they might be interested in your journey here. I run armbian on both of these SBCs, and they have been rock solid.

**Comment by vasvir** [comment]
Congrats. What a journey! Very nicely written too.
